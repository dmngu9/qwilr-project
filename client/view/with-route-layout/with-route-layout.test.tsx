import * as React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { Route } from 'react-router-dom';

import withRouteLayout from './with-route-layout';

interface Props {
    children?: React.ReactNode;
}

const TestLayout: React.StatelessComponent<Props> = ({ children }) => <div>{children}</div>;

const TestComponent: React.StatelessComponent = () => <div>test component</div>;

describe('withRouteLayout', () => {
    let wrapper: ShallowWrapper;

    beforeEach(() => {
        const RouteLayout = withRouteLayout(TestLayout);
        wrapper = shallow(<RouteLayout path="/" component={TestComponent} />);
    });

    it('should render correctly as Route', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should pass in correct component to Route', () => {
        const Component = wrapper.find(Route).props().component;

        expect(shallow(<Component />)).toMatchSnapshot();
    });
});
