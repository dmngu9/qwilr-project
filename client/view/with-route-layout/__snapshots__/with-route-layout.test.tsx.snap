// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withRouteLayout should pass in correct component to Route 1`] = `
<TestLayout>
  <TestComponent />
</TestLayout>
`;

exports[`withRouteLayout should render correctly as Route 1`] = `
<Route
  component={[Function]}
  path="/"
/>
`;
